lib/bugsnag/api/client.rb

```diff
@@ -169,9 +171,9 @@ def agent
           http.headers[:user_agent] = configuration.user_agent
 
           if basic_authenticated?
-            http.basic_auth configuration.email, configuration.password
+            http.request :authorization, "Basic", Base64.strict_encode64("#{configuration.email}:#{configuration.password}")
```

Comment:
As far as I can tell, this is backwards compatible at least as far back as Faraday `0.9.0`, and forward compatible with Faraday `2.0.0`, where as 

```ruby
http.request :authorization, "Basic", configuration.email, configuration.password
```

would be compatible with Faraday `2`, but **not** backwards compatible with Faraday `< 2`.